R-package/R/lgb.Booster.R

```diff
@@ -799,18 +811,36 @@ predict.lgb.Booster <- function(object,
     ))
   }
 
-  return(
-    object$predict(
-      data = newdata
-      , start_iteration = start_iteration
-      , num_iteration = num_iteration
-      , rawscore = rawscore
-      , predleaf =  predleaf
-      , predcontrib =  predcontrib
-      , header = header
-      , params = params
-    )
+  type <- head(type, 1L)
```

Comment:
What is the purpose of having `type` default to a vector with 5 elements, and then always taking only the first thing provided? It seems to me that it would be simpler to just default to `"link"`.

Reply:
The purpose is to have the allowed values visible in the function signature so that they are easily seen by the user and easy to autocomplete, in the same way as for example base R's `predict.glm`.